### DISCUSSION TITLE
Forcing data about ERA5

### DISCUSSION THREAD

**Discussion by Zhoujieliang** [NCAR/hrldas#233] [Q&A] [answered]
I have successfully create the forcing data according to the 'README.ERA5' steps(https://github.com/NCAR/hrldas/blob/master/hrldas/docs/README.ERA5).
But in web(https://cds.climate.copernicus.eu/cdsapp#!/dataset/reanalysis-era5-land?tab=overview), when download ERA5_Land, it seems that only one month of data can be downloaded at most, and it cannot be downloaded for more than 31 days.
![image](https://github.com/user-attachments/assets/a38a9692-9c7c-4840-9321-c3899a22183b)
Is there a good way to create one-year forcing data?

**Comment by tslin2** [comment]
You could try using CDSAPI
https://cds.climate.copernicus.eu/how-to-api

**Comment by sudarsan3010** [comment, marked as answer]
Dear @Zhoujieliang,

Even through CDSAPI, the request can be placed only for one month. But you can loop the requests in with simple Python code.
Attached herewith.
[download_era5land.txt](https://github.com/user-attachments/files/18194976/download_era5land.txt)
(In the above file just change your Python library path and make it a .py executable file)

This Python code will download the ERA5_Land forcings months, which can later be merged together by the following command.
`cdo -O cat ERA5_global_2008*.grib ERA5_global_2008.grib`


(It is necessary to have both ecCodes and cdo (with ecCodes support) installed on the system)

**Reply by Zhoujieliang** [reply]
thanks a lot, it very help for me.
